cmd/cloudshell_open/artifactregistry.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	artifactregistry "cloud.google.com/go/artifactregistry/apiv1"
+	artifactregistrypb "cloud.google.com/go/artifactregistry/apiv1/artifactregistrypb"
+)
+
+// Create a "Cloud Run Source Deploy" repository in Artifact Registry (if it doesn't already exist)
+func createArtifactRegistry(project string, region string, repoName string) error {
+
+	repoPrefix := fmt.Sprintf("projects/%s/locations/%s", project, region)
+	repoFull := fmt.Sprintf("%s/repositories/%s", repoPrefix, repoName)
+
+	ctx := context.Background()
+
+	//Check for existing repo
+	client, err := artifactregistry.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create artifact registry client: %w", err)
+	}
+
+	req := &artifactregistrypb.GetRepositoryRequest{
+		Name: repoFull,
+	}
+	existingRepo, err := client.GetRepository(ctx, req)
+
+	if err != nil {
+		// TODO need to continue if error is "code = NotFound desc = Requested entity was not found."
```

Comment:
I need to learn how to check for this specific error. 

"code = NotFound desc = Requested entity was not found." is returned if the registry doesn't exist, which is an ok error to get, and should force the creation of the registry.

Reply:
@grayside or @subfuzion any tips? 
This check may replace the `existingRepo == nil` check later (either we allow this "NotFound" error now and continue, or create if this error is encountered)

Reply:
This might be helpful to you in this situation. Take a look at `errors.Is` (https://pkg.go.dev/errors).

Reply:
I used this, but had to build a grpc error entity to validate against (used https://pkg.go.dev/google.golang.org/grpc/status#Error and https://pkg.go.dev/google.golang.org/grpc/codes)